projects/filjs-audit/README.md

```diff
@@ -0,0 +1,391 @@
+# Lotus / Filecoin RPC and JavaScript Library Audit
+
+**2021-04**
+
+This report captures a condensed summary of a brief audit conducted of the current state of the JavaScript ecosystem surrounding Filecoin and the affordances provided to JavaScript developers via the Lotus RPC API.
+
+## Lotus RPC API
+
+### OpenRPC support
+> JSON based API definition of the lotus json-rpc interfaces that are auto-generated from the code.
+
+The Good: 
+- it is genearted from the go rpc implementation, so is always up to date. 
```

Comment:
```suggestion
- it is generated from the go rpc implementation, so is always up to date. 
```

from my experience playing with this auto generated typescript declarations, there is a pain point on optionality. Interfaces are created taking into account that all the parameters are required and if you use this within a typescript project, the compiler will complain that you need to provide all the things, while some of them are optional